lua/telescope/_extensions/file_browser/actions.lua

```diff
@@ -417,6 +417,37 @@ fb_actions.goto_cwd = function(prompt_bufnr)
   current_picker:refresh(finder, { reset_prompt = true, multi = current_picker._multi })
 end
 
+--- Change working directory of nvim to the selected file/folder in |builtin.file_browser|.
+---@param prompt_bufnr number: The prompt bufnr
+fb_actions.change_cwd = function(prompt_bufnr)
+  local picker = action_state.get_current_picker(prompt_bufnr)
+  local finder = picker.finder
+  local sel = fb_utils.get_selected_files(prompt_bufnr, true)[1]
+  finder.path = sel:is_dir() and sel:absolute() or sel:parent():absolute()
+  finder.cwd = finder.path
+  finder.files = true
```

Comment:
Something seems off here. If triggered from `folder_browser`, this changes to `file_browser`.

I wouldn't make any assumptions about what the user is doing because the user can just switch to `file_browser`, no?

In any case, prompt and results title would most likely be inappropriately be handled right now.

Reply:
So you suggest I remove L428? Don't really understand what should we expect...

Also, why shouldn't we change the title there? This way users can see that the cwd directory has chenged as the title reflects the relative path.

Reply:
The `action` as is forces a change to `file_browser` mode (`finder.files = true` and `picker:refresh`).

Since we've now agreed that this `action` also has valid usage from `folder_browser` mode, the lines I've referred to induce forced behavior difficult to anticipate whether to user wants or doesn't want.

If we do this as is from `folder_browser`, we'd be forced into `file_browser` mode while not having the `prompt_title` changed (which reminds me, I should probably factor title handling into a `fb_util` or tie it somehow more intricately to `finder`).

So I guess it's cleanest to just refresh the current `file` or `folder_browser` (with titles updated, I guess as is is fine) without forcing the change to `file_browser`.